### PR TITLE
Add option to limit pathing to a given tile image

### PR DESCRIPTION
**Pull request by eanders-ms**
Demo project: https://arcade.makecode.com/59709-50652-26659-16628
(Run around and press A to spawn a pathing sprite)

I wanted a-star pathing but didn't want to add walls for other gameplay reasons, so I added an optional parameter to a-star to pass in a walkable tile image. Lemme know what'cha think.

### CODE REVIEW

File: a-star.ts [hunk 1]
```diff
@@ -221,4 +223,11 @@ namespace scene {
         const c = locationCol(l);
         return tm.isObstacle(c, r);
     }
+
+    function isWalkable(l: tiles.Location, onTilesOf: Image, tm: tiles.TileMap): boolean {
+        if (isWall(l, tm)) return false;
+        if (!onTilesOf) return true;
+        const img = tm.getTileImage(tm.getTileIndex(l.col, l.row))
+        return img == onTilesOf;
```
Comment (jwunderl) [inline a-star.ts]:
```suggestion
        return img.equals(onTilesOf);
```

in case they pass in their own redrawn image instead of the const?

**Review by jwunderl — APPROVED** [review]
Ooh nice LGTM, tagging @darzu as well as I think I remember you wanting this feature? ~

File: a-star.ts (hunk 1 above)
Comment (eanders-ms) [reply a-star.ts]: Agreed & fixed.

**Comment by jwunderl** [comment]
Merged and bumped, thanks!

**Comment by eanders-ms** [comment]
Turns out this change introduced a compatibility issue.  When I updated test.ts a tilemap.g.ts was generated, and when imported into a project, the tile resources conflict. Quick fix would be to remove the test code, but this seems like a problem in need of a more general solution. Extensions should be able to contain tilemaps.

IMG_4382

**Comment by jwunderl** [comment]
right, for some reason I had thought we had handled that case when adding the new `tilemap.g.ts` implementation but I think we just discussed it instead / clearly there would have to be some better renaming logic for that to make any sense. Fixed by bumping those files to be `testFiles` so they only get loaded when the extension is the top level project: https://github.com/jwunderl/arcade-tilemap-a-star/blob/master/pxt.json#L15-L18 and bumped to v0.3.1, but we'll need a proper fix in pxt-core eventually too

**Comment by eanders-ms** [comment]
Working better now, thank you! I'll open an issue in the pxt repo to track this.

**Pull request merged**